launch/dynamo-run/src/subprocess/vllm_inc.py

```diff
@@ -170,14 +163,18 @@ async def init(runtime: DistributedRuntime, config: Config):
         arg_map = {**arg_map, **json_map}  # json_map gets precedence
 
     # Patch won't start KVCacheEventManager unless these four are set
-    os.environ["VLLM_WORKER_ID"] = str(endpoint.lease_id())
-    os.environ[
-        "VLLM_KV_CAPI_PATH"
-    ] = "libdynamo_llm_capi.so"  # Must be on LD_LIBRARY_PATH
-    os.environ["VLLM_KV_NAMESPACE"] = config.namespace
-    os.environ["VLLM_KV_COMPONENT"] = config.component
-
-    os.environ["VLLM_NO_USAGE_STATS"] = "1"  # Avoid internal HTTP requests
+
+    component = runtime.namespace(config.namespace).component(config.component)
+    await component.create_service()
+    endpoint = component.endpoint(config.endpoint)
+
+    os.environ.setdefault("VLLM_WORKER_ID", str(endpoint.lease_id()))
+    os.environ.setdefault(
+        "VLLM_KV_CAPI_PATH", "libdynamo_llm_capi.so"
+    )  # Must be on LD_LIBRARY_PATH
+    os.environ.setdefault("VLLM_KV_NAMESPACE", config.namespace)
+    os.environ.setdefault("VLLM_KV_COMPONENT", config.component)
```

Comment:
Do these two (VLLM_KV_NAMESPACE, VLLM_KV_COMPONENT) make sense to be set externally by user when correct config namespace/endpoint are known here in the program? 

Seems like best case scenario is user sets it correctly (same as always setting it without setdefault here), worst case they set it incorrectly  and something doesn't work as expected?